workspace_tools/export/gcc_arm_common.tmpl

```diff
@@ -1,12 +1,38 @@
 # This file was automagically generated by mbed.org. For more information, 
 # see http://mbed.org/handbook/Exporting-to-GCC-ARM-Embedded
 
+ifeq ($(OS),Windows_NT)
+MKDIR := mkdir
+else
+MKDIR := mkdir -p
+endif
+
+ifeq (,$(filter bin,$(notdir $(CURDIR))))
+.SUFFIXES:
+OBJDIR := bin
+MAKETARGET = $(MAKE) --no-print-directory -C $@ -f $(CURDIR)/Makefile \
```

Comment:
This seems a bit hacky. Would it be better to add the bin/ prefix to your OBJECTS and switch to pattern based rules for compiling/assembling that take into account the slight difference in pathname? That way you don't need to recursively call make?
```
$(OBJDIR)/%.o: %.s
 	@[ -d $(dir $@)] || $(MKDIR) $(dir $@)
  	$(CC) $(CPU) -c -x assembler-with-cpp -o $@ $<
```